### PR TITLE
Expand const args with globals

### PR DESCRIPTION
**Pull request by bpkroth**
Builds off of #533 

Allows for const_args of the form:

```json
"const_args": {
  "DB_DIR": "$workdir/db",
  "DB_BAK_DIR": "$workdir/db.bak",
}
```

Which can be expanded based on the value of `$workdir` from the `globals` config (which may in turn be expanded from environment variables per prior PRs).

### CODE REVIEW

File: mlos_bench/mlos_bench/environments/base_environment.py [hunk 1]
```diff
@@ -172,6 +174,19 @@ def _expand_groups(groups: Iterable[str],
...
+        """
+        Expand `$var` into actual values of the variables.
+        """
+        res: Dict[str, TunableValue] = {}
+        for key, val in params.items():
+            if isinstance(val, str):
+                res[key] = Template(val).safe_substitute(global_config)
+            else:
+                res[key] = val
+        return res
```
Comment (motus) [inline mlos_bench/mlos_bench/environments/base_environment.py:188]: maybe something like
```suggestion
        return {
            key: Template(val).safe_substitute(global_config) if isinstance(val, str) else val
            for key, val in params.items()
        }
```

File: mlos_bench/mlos_bench/launcher.py [hunk 2]
```diff
@@ -257,8 +257,8 @@ def _expand_vars(self, value: Any) -> Any:
             # new set of params explicitly.
             # Since this operates by updating the global_config along the way,
             # we need to continually update the set of params to use for substitution.
-            params = dict(os.environ)
-            params.update(self.global_config)
+            params = self.global_config.copy()
+            params.update(dict(os.environ))
```
Comment (motus) [inline mlos_bench/mlos_bench/launcher.py:261]: just
```suggestion
            params.update(os.environ)
```
if mypy doesn't mind

**Review by motus — APPROVED** [review]
looks good! left a few minor comments

File: mlos_bench/mlos_bench/launcher.py (hunk 2 above)
Comment (bpkroth) [reply mlos_bench/mlos_bench/launcher.py]: Answered in #533

**Pull request merged**